src/components/tab/FileTabs.vue

```diff
@@ -31,34 +31,114 @@
 </template>
 
 <script setup>
-import { ref, watch } from 'vue';
+import {
+  ref,
+  watch,
+  onMounted,
+  onUnmounted,
+} from 'vue';
 import FileTabHeader from 'src/components/tab/FileTabHeader.vue';
-
-const emit = defineEmits(['update:modelValue', 'update:close-file']);
+import FileEvent from 'src/composables/events/FileEvent';
 
 const props = defineProps({
-  files: {
-    type: Array,
-    required: true,
-  },
-  modelValue: {
+  fileInformations: {
     type: Object,
     required: true,
   },
 });
 
-const activeFileId = ref(props.modelValue.id);
+const fileTabArray = ref([]);
+const activeFileId = ref(null);
 
-watch(activeFileId, () => {
-  if (activeFileId.value !== props.modelValue.id) {
-    emit('update:modelValue', { isSelected: true, id: activeFileId.value });
+let selectFileNodeSubscription;
+
+/**
+ * Set activeFileId equal to the id.
+ * @param {String} id - Id of selected file.
+ */
+function setActiveFileId(id) {
```

Comment:
This function and the same in previous commit, I don't understand why we have it. In previous commit you use it, but here you don't use it each time you can. So, I think we have to discuss about this.

Reply:
can you name the two previously mentioned commits ? there are a lot...

Reply:
done